### PR TITLE
Switching to NPM scripts

### PR DESCRIPTION
**Pull request by pateketrueke**
cc: @ducin 

- [x] ~~I'm still thinking about related stuff with distributions, locales and such.~~
- [x] ~~I guess using only npm-scripts we wont't solve that, can we try using the browserify api for that?~~

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -4,7 +4,10 @@ branches:
     - gh-pages
 node_js:
   - 0.10
-before_script:
-  - 'npm install -g grunt-cli coveralls'
+  - 0.12
+  - 4
```
Comment (ducin) [inline .travis.yml]: oh, yeah, that's VERY important!

File: .banner.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 /*!
  * <%= pkg.name %> library v<%= pkg.version %>
  * <%= pkg.homepage %>
+ * @preserve
```
Comment (ducin) [inline .banner.txt]: how does that work, what will it do?

File: lib/types/number.js [hunk 3]
```diff
@@ -1,5 +1,3 @@
-'use strict';
-
 var MIN_INTEGER = -100000000,
     MAX_INTEGER = 100000000,
     DEFAULT_PRECISION   = .00001;
```
Comment (ducin) [inline lib/types/number.js]: how about adding 0 in front of .00001? I know it's valid JS syntax, but 0.00001 is hack-less and looks cleaner for me.

File: lib/index.js [hunk 4]
```diff
@@ -1,5 +1,3 @@
-'use strict';
-
```
Comment (ducin) [inline lib/index.js]: why did you remove `'use strict';` everywhere? I know we will build our project in a different way - but how does that relate to source code being strict?

File: package.json [hunk 5]
```diff
@@ -4,6 +4,17 @@
   "description": "JSON-Schema + Faker",
   "homepage": "http://json-schema-faker.js.org",
   "main": "lib/index.js",
+  "scripts": {
```
Comment (ducin) [inline package.json]: well, I never used npm-scripts in favor of grunt, but let's give it a try.
I must say I still don't clearly understand what `grunt parts` does so I hope this one would be cleaner for me. I will let you know if I see any problems with npm scripts.

**Comment by ducin** [comment]
Thanks for this PR. I think it's better to do a PR in code - then we can do *peer review*! Please comment on those and let's see where we are then.

I know that the minification of our bundle was crappy, it was not realy minified - but I guess your patch might have solved that issue.

File: .banner.txt (hunk 2 above)
Comment (pateketrueke) [inline .banner.txt]: without this annotation `uglify-js` will strip the comment, see: https://github.com/mishoo/UglifyJS2#keeping-copyright-notices-or-other-comments

File: lib/types/number.js (hunk 3 above)
Comment (pateketrueke) [inline lib/types/number.js]: I didn't do it! :beers:

...of course seems bit ugly +1 for `0.00001`

File: .banner.txt (hunk 2 above)
Comment (ducin) [inline .banner.txt]: didn't know that one, makes sense!

PS isn't it 3:53 your time, what the hell are you doing on github at that time?! :)

File: lib/index.js (hunk 4 above)
Comment (pateketrueke) [inline lib/index.js]: Updating to latest ESLint made the `'use strict'` appear as warning, just to mitigate that.

Of course we had no benefit from using (if we follow standards) it and we lose nothing without.

Is just that "the future" is strict by default, I like that. :beers:

See also:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode
- http://eslint.org/docs/rules/#strict-mode

File: package.json (hunk 5 above)
Comment (pateketrueke) [inline package.json]: That horrible frankenstein called `grunt-parts` was a former way of thinking about automating... sorry for that.

From now on I'll try to use npm-scripts for everything.

Please give it a try. :+1:

**Comment by pateketrueke** [comment]
> I know that the minification of our bundle was crappy, it was not realy minified - but I guess your patch might have solved that issue.

Of course my horse!

Run `npm update` and then `npm run build`, you'll get both files as expected.

Also check out the `build-dist.js` script, there's a draft for what I've thinking about locales, set `withLocales = true` and execute `npm run build` again to see the results.

Of course that's a browserify hack... I think we do support the CommonJS specification just as Faker.js does for `require('json-schema-faker/locale/es_MX')` calls.

If we do that later we can refactor our build task to use `browserify('./locale/es_MX.js')` etc.

What do you think?

File: .banner.txt (hunk 2 above)
Comment (pateketrueke) [inline .banner.txt]: See ya later, I'll take nap! :zzz:

File: package.json (hunk 5 above)
Comment (ducin) [inline package.json]: Sure I will!
Last time you tried to convince me to use `grunt parts` ;)

**Comment by pateketrueke** [comment]
I've been working on the `npm run build` task and I think I got it:

## Locale support & bundling

The build script create proxy files using faker's locales and write them out to, e.g. `./locale/es_MX.js` (same way as faker.js does)

This way you can rely on `require('json-schema-faker/locale/es_MX')` and so on.

## Distribution files

Also, the script will bundle all generated scripts to `./dist/` so it can be served as well by cdnjs, bower, etc.

In this topic @ducin is required to include minified versions from all locales too for cdnjs?

With this, I think we're now able to close #91 #90 and #56 :beers:

**Comment by ducin** [comment]
@pateketrueke regarding our branch and browserify bundling:
1. we shall move build scripts out of the main folder. Keeping `build-dist.js` as well as `.locale.js` and my `.banner.txt` is introducing some mess - and I guess there'd be more files like that. I'd move them all to a separate dir for clarity - and update `package.json`.scripts accordingly.
2.  `withLocales = true` - is it by default or is it just hardcoded and you can't change it?
